### ISSUE TITLE
package.json publishConfig is not applied

### ISSUE THREAD

**Issue by boromisp** [wclr/yalc#158]
It would be awesome if at least the `main` and `module` values could be overwritten based on the `publishConfig`:

https://yarnpkg.com/configuration/manifest/#publishConfig

https://classic.yarnpkg.com/en/docs/package-json#toc-publishconfig

https://docs.npmjs.com/cli/v6/configuring-npm/package-json#publishconfig

The yarn v2 implementation seems to be stricter in what config values can be overwritten.

**Comment by wycats** [comment]
I totally agree. I got everything working with yalc but this is a blocker for me 😦

**Comment by wclr** [comment]
How should `publishConfig` be applied?

**Comment by corbt** [comment]
The values in `publishConfig` should overwrite the relevant top-level values in the base `package.json`. Here's the documentation for how pnpm does it: https://pnpm.io/package_json#publishconfig

**Comment by wycats** [comment]
(Sorry for not replying earlier, but @corbt got it).

Circling back around, this is still a blocker for me. It creates a divergence between `pnpm publish` and `yalc publish` that breaks the whole concept of the `yalc` workflow for me.

Maybe you could consider using `pnpm package` directly, which would guarantee equivalence?

**Comment by NullVoxPopuli** [comment]
> pnpm package

`pnpm pack`*

**Comment by zhangfisher** [comment]
use `yald` , it is fork yalc,add `publishConfig` support

[github](https://github.com/zhangfisher/yald)

**Comment by AlanSl** [comment]
`pnpm pack` creates a realistic package bundle including applying publishConfig to the package.json (and may apply other not-yet-supported pnpm prepublish actions too), so running a command like this in the workspace you want to publish/push could be a temporary workaround:

```sh
file=$(pnpm pack); tar -xvzf $file; rm $file; cd package; yalc publish; yalc push; cd ..; rm -rf package;
```

This generates a tarball of what would be published to the registry (applying publishConfig among other things), notes the filename, unzips it, steps inside it, does `yalc publish` and `yalc push` in this more-realistic unzipped package bundle, steps out again, then deletes it.

---

and I agree with @wycats that it'd be good if Yalc either directly used `pack` under the hood, or had an option to use it, so it natually stays in sync with what the package manager would actually do to create the published bundle.

**Comment by xdewx** [comment]
> use `yald` , it is fork yalc,add `publishConfig` support
> 
> [github](https://github.com/zhangfisher/yald)

why not a PR ?
